wit/deps/observe/world.wit

```diff
@@ -0,0 +1,8 @@
+package wasi:observe@0.2.0-draft;
+
+world imports {
+    import tracer;
+}
+
+// TODO(Reviewer): Should we make this an experimental wasi package or a Spin package
```

Comment:
I would suggest that we land WASI Observe as a release candidate `wasi:observe@0.2.2-rc2024-09-17`. I think it is the right choice to put it in the WASI namespace b/c WASI Observe is a [phase 1 proposal](https://github.com/WebAssembly/wasi-observe/pull/15).

If we're feeling conservative we could add this to a new `http-trigger-experimental` world where we make less support guarantees.

```wit
package fermyon:spin@2.0.0;

/// The full world of a guest targeting an http-trigger
world http-trigger {
  include platform;
  export wasi:http/incoming-handler@0.2.0;
}

/// Like `http-trigger`, but using WASI 0.2.0-rc-2023-10-18
world http-trigger-rc20231018 {
  include platform-rc20231018;
  export wasi:http/incoming-handler@0.2.0-rc-2023-10-18;
}

/// Like `http-trigger`, but experimental and with no support guarantees
world http-trigger-experimental {
  include platform-experimental;
  export wasi:http/incoming-handler@0.2.0;
}

/// The imports needed for a guest to run on a Spin host
world platform {
  include wasi:cli/imports@0.2.0;
  import wasi:http/outgoing-handler@0.2.0;
  import llm;
  import redis;
  import mqtt;
  import postgres;
  import mysql;
  import sqlite;
  import key-value;
  import variables;
}

/// Like `platform`, but using WASI 0.2.0-rc-2023-10-18
world platform-rc20231018 {
  include wasi:cli/reactor@0.2.0-rc-2023-10-18;
  import wasi:http/outgoing-handler@0.2.0-rc-2023-10-18;
  import llm;
  import redis;
  import mqtt;
  import postgres;
  import mysql;
  import sqlite;
  import key-value;
  import variables;
}

/// Like `platform`, but experimental and with no support guarantees
world platform-experimental {
  include platform;
  include wasi:observe/imports@0.2.2-rc2024-09-17;
}
```

Reply:
(This is probably a topic deserving of a SIP)

I think we should use WIT's [feature gates](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#feature-gates) for this, like so:

```wit
world http-trigger {
  include platform;
  export wasi:http/incoming-handler@0.2.0;
  @unstable(feature = wasi-observe)
  include wasi:observe@0.2.0-draft-2024-09-20/imports;
}
```

Additionally, unstable features should require runtime opt-in with something like a CLI flag along the lines of `--enable-unstable-wit-features=wasi-observe,...`

Reply:
Yes: sloppiness. It absolutely should be in `platform`